### DISCUSSION TITLE
Conditional Portfolio Allocation Constraints

### DISCUSSION THREAD

**Discussion by Dominik-Cisar** [dcajasn/Riskfolio-Lib#179] [Q&A]
Is it possible to implement conditional portfolio constraints, such as setting a minimum allocation of 5% only if an asset is selected in the portfolio? If not, are there plans to add this feature?

For example I'm modelling funds of funds portfolio where underlying funds are divided into 2 groups - Market neutral and Directional. To limit allocation per group is easy. Also, it's easy to set max allocation to fund - 20% of portfolio... but is there a possibility to constrain the minimum allocation which cannot be less than 5%? It cannot be done like upper bound of 20% because It'll set all allocations to 5% which I don't want.
Here are the constraints with the experimental lower bound which obviously did not worked🤣 so I disabled it.
```
constraints = {'Disabled':      [False, False, False, True, False, False, False, False, False, False, False, False, False, False, False, False],
               'Type':          ['Classes', 'Classes', 'All Assets', 'All Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets', 'Assets'],
               'Set':           ['Group', 'Group', '', '', '', '', '', '', '', '', '', '', '', '', '', ''],
               'Position':      ['Market neutral', 'Directional', ''   , ''  , 'X0_003a', 'X705a', 'X711a', 'X711b','X755', 'X867', 'X486a', 'X883' ,'X574a', 'X574b', 'X574c', 'X705b'],
               'Sign':          ['<='            , '<='         , '<=' , '>=', '>='     , '>='   , '>='   , '>='   , '<=' , '>='  , '<='   , '<='   , '<='  , '<='   , '<='   , '<='   ],
               'Weight':        [0.30            , 0.70         , 0.20 , 0.05, 0.10     , 0.15   , 0.05   , 0.15   , 0.05 , 0.10  , 0.00   , 0.00   , 0.00  , 0.00   , 0.00   , 0.00   ],
               'Type Relative': ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', ''],
               'Relative Set':  ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', ''],
               'Relative':      ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', ''],
               'Factor':        ['', '', '', '', '', '', '', '', '', '', '', '', '', '', '', ''],}
#make DF out of it
constraints = pd.DataFrame(constraints)
#convert asset constraints to numbers
A, B = rp.assets_constraints(constraints, asset_classes)
port.ainequality = A
port.binequality = B
```

**Comment by dcajasn** [comment]
Hi @Dominik-Cisar 

The kind of constraint that you ask is something that is more advanced and it is not implemented in Riskfolio-Lib. Also, I don't plan to add that feature in Riskfolio-Lib.

Best,
Dany

**Reply by Dominik-Cisar** [reply]
Hi @dcajasn, 

thanks very much for the answer and for the effort on the library. I get it, it is a nice to have feature which could be solver quite easily after portfolio optimization. So not really important.

Thanks

Dominik
